### PR TITLE
Charlesmchen/holiday code review omnibus

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

**Comment by charlesmchen** [comment]
See: https://github.com/WhisperSystems/Signal-iOS/pull/2226

File: src/Messages/OWSMessageSender.m [hunk 1]
```diff
@@ -212,18 +210,17 @@ - (instancetype)initWithMessage:(TSOutgoingMessage *)message
...
-        }
-
         [strongSelf.message updateWithSendingError:error];
 
         DDLogDebug(@"%@ failed with error: %@", strongSelf.tag, error);
         aFailureHandler(error);
-        [strongSelf markAsComplete];
+
+        // Ensure we call the success or failure handler exactly once.
+        @synchronized(strongSelf)
```
Comment (michaelkirk) [inline src/Messages/OWSMessageSender.m]: nit: I think we could just do the check *in* `markAsComplete` and dedupe the two asserts blocks.

**Review by michaelkirk — APPROVED** [review]
LGTM

File: src/Messages/OWSMessageSender.m (hunk 1 above)
Comment (charlesmchen) [reply src/Messages/OWSMessageSender.m]: Done.

**Pull request merged**